zhaquirks/tuya/air/__init__.py

```diff
@@ -31,9 +31,42 @@ class TuyaAirQualityVOC(TuyaLocalCluster):
     client_commands = {}
 
 
+class CustomTemperature(t.Struct):
+    """Custom temperature wrapper."""
+
+    field_1: t.int16s_be
+    temperature: t.int16s_be
+
+    @classmethod
+    def from_value(cls, value):
+        """Convert from a raw value to a Struct data."""
+        return cls.deserialize(value.serialize())[0]
+
+
 class TuyaAirQualityTemperature(TemperatureMeasurement, TuyaLocalCluster):
     """Tuya temperature measurement."""
 
+    attributes = TemperatureMeasurement.attributes.copy()
+    attributes.update(
+        {
+            # ramdom attribute IDs
+            0xEF12: ("custom_temperature", CustomTemperature, False),
+        }
+    )
+
+    def update_attribute(self, attr_name: str, value: Any) -> None:
+        """Calculate the current temperature."""
+
+        super().update_attribute(attr_name, value)
+
+        if attr_name == "custom_temperature":
+            custom_temperature = self._attr_cache.get(
+                self.attributes_by_name[attr_name].id,
+            )
```

Comment:
Not that it matters, but what's the benefit of getting `custom_temperature` from the attribute cache again when it should also be `value` here?

Reply:
You are right!!! 
I was expecting that value was the raw value (so no `value.temperature`) and we have to recover from class 😅 
Just checked and working like a charm

Thanks Julian!